hashtable/hashmap.go

```diff
@@ -0,0 +1,161 @@
+package hashtable
+
+const (
+	// KeyNotFound is a constant indicates that
+	// a key cannot be found in the HashMap.
+	KeyNotFound     = 0
```

Comment:
`KeyNotFound` should be a rarely used value. I prefer to using min value of `int` type